lib/Db/TimelineQueryFaceRecognition.php

```diff
@@ -0,0 +1,222 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OCA\Memories\Db;
+
+use OCP\DB\QueryBuilder\IQueryBuilder;
+use OCP\Files\Folder;
+use OCP\IDBConnection;
+
+trait TimelineQueryFaceRecognition
+{
+    protected IDBConnection $connection;
+
+    public function transformFaceRecognitionFilter(IQueryBuilder &$query, string $userId, string $personStr)
+    {
+        // Get title and uid of face user
+        $personNames = explode('/', $personStr);
+        if (2 !== \count($personNames)) {
+            throw new \Exception('Invalid person query');
+        }
+
+        $personUid = $personNames[0];
+        $personName = $personNames[1];
+
+        // Join with images
+        $query->innerJoin('m', 'facerecog_images', 'fri',
+            $query->expr()->eq('fri.file', 'm.fileid')
```

Comment:
I'm yet to test this, but I asssume `fri.file` is `BIGINT(20)`? Otherwise this is gonna be horribly slow.

Reply:
Ohhh.. Good catch..
I never realized it. I will add a migration in my application to adapt it. 😉

Reply:
Done on https://github.com/matiasdelellis/facerecognition/pull/608/commits/bfdaab2cd24c58b737c65351e02493fc33055d54